src/ReloadIntegration/ReloadHostingStartup.cs

```diff
@@ -22,16 +25,20 @@ public void Configure(IWebHostBuilder builder)
             }
             ClearStartupHookEnvironmentVariable();
 
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) || RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
             {
-                uint listenFd;
+                uint listenHttpFd, listenHttpsFd;
                 try
                 {
-                    var listenFdAsString = Environment.GetEnvironmentVariable("ZOCKET_LISTEN_FD");
-                    listenFd = uint.Parse(listenFdAsString);
-                    builder.ConfigureKestrel(options =>
+                    listenHttpFd = uint.Parse(Environment.GetEnvironmentVariable("ZOCKET_LISTEN_HTTP_FD"));
+                    listenHttpsFd = uint.Parse(Environment.GetEnvironmentVariable("ZOCKET_LISTEN_HTTPS_FD"));
+                    builder.ConfigureServices(services =>
                     {
-                        options.ListenHandle(listenFd);
+                        services.PostConfigure<KestrelServerOptions>(options =>
+                        {
+                            options.ListenHandle(listenHttpFd);

```

Comment:
Did you manage to check if this was additive instead of replace?

Reply:
It is additive